src/core_plugins/kibana/public/dashboard/dashboard.js

```diff
@@ -176,9 +185,14 @@ app.directive('dashboardApp', function ($injector) {
         }
       };
 
-      $scope.filterResults = function () {
-        dashboardState.applyFilters($scope.model.query, filterBar.getFilters());
-        $scope.refresh();
+      $scope.updateQuery = function (query) {
```

Comment:
yea, although sounds like a lot of this logic will be extracted out with the follow up PR, so I think it's okay to can wait till then and see what/if any logic we can push into dashboardState.